### PR TITLE
K5Tests

### PR DESCRIPTION
**Pull request by ndrsszsz**
[Bitrise - Canvas iOS - Student - E2ETests](https://app.bitrise.io/build/003e1db2-15d6-4e70-9017-4f4de0ae3880)

4 new K5 test cases (olds are deleted)

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please add only one release note.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Please add only one affects.</td>
    </tr>
  </tbody>
</table>



#### Release Note: 
None

#### Affected Apps: None

[MBL-16946](https://instructure.atlassian.net/browse/MBL-16946)
[MBL-16771](https://instructure.atlassian.net/browse/MBL-16771)
[MBL-16848](https://instructure.atlassian.net/browse/MBL-16848)
[MBL-16772](https://instructure.atlassian.net/browse/MBL-16772)
[MBL-16808](https://instructure.atlassian.net/browse/MBL-16808)
[MBL-16816](https://instructure.atlassian.net/browse/MBL-16816)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 92.42% | 92.42% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f7bd6794f88098907c9bd3698bd6d362e20d1bbf
</p>

File: TestsFoundation/TestsFoundation/DataSeeding/Course/DSCourse.swift [hunk 1]
```diff
@@ -23,4 +23,6 @@ public struct DSCourse: Codable {
     public var syllabus_body: String?
     public let account_id: String
     public var homeroom_course: Bool = false
+    public let start_at: String?
+    public let end_at: String?
```
Comment (vargaat) [inline TestsFoundation/TestsFoundation/DataSeeding/Course/DSCourse.swift:27]: Why not dates?

File: TestsFoundation/TestsFoundation/Extensions/XCUIElementExtensions.swift [hunk 2]
```diff
@@ -42,6 +42,7 @@ public extension XCUIElement {
         case swipeDown
         case swipeRight
         case swipeLeft
+        case selfSwipeLeft
```
Comment (vargaat) [inline TestsFoundation/TestsFoundation/Extensions/XCUIElementExtensions.swift:45]: Maybe adding an associated value to the `swipeLeft` case would be more elegant. This associated value could have a type name `Target` with 2 values: `onApp`, `onElement`, you could make it as an inner enum within the `ElementAction` enum.

File: TestsFoundation/TestsFoundation/Helpers/GradesHelper.swift [hunk 3]
```diff
@@ -28,9 +28,10 @@ public class GradesHelper: BaseHelper {
     public static func gradeOutOf(assignment: DSAssignment? = nil,
                                   assignmentId: String? = nil,
                                   actualPoints: String,
-                                  maxPoints: String) -> XCUIElement {
+                                  maxPoints: String, letterGrade: String = "") -> XCUIElement {
```
Comment (vargaat) [inline TestsFoundation/TestsFoundation/Helpers/GradesHelper.swift:31]: Not something significant but other parameters are all on a new line, let's follow the patter with this one.

File: TestsFoundation/TestsFoundation/Helpers/GradesHelper.swift [hunk 4]
```diff
@@ -28,9 +28,10 @@ public class GradesHelper: BaseHelper {
...
         let assignment = app.find(id: "GradeListCell.\(assignment?.id ?? assignmentId!)")
-        return assignment.find(label: "Grade, \(actualPoints) out of \(maxPoints)")
+        let lgSuffix = letterGrade != "" ? " (\(letterGrade))" : letterGrade
```
Comment (vargaat) [inline TestsFoundation/TestsFoundation/Helpers/GradesHelper.swift:33]: Inverting the condition would make it easier to understand.
```suggestion
        let lgSuffix = letterGrade == "" ? "" : " (\(letterGrade))"
```

**Review by vargaat — CHANGES_REQUESTED** [review]

**Review by idevainstr — APPROVED** [review]

**Pull request merged**